engine/configurator/conf_yaml.c

```diff
@@ -370,364 +551,586 @@ parse_config_yaml_dependency_entry(yaml_document_t *d,
     te_errno        rc;
 
-    if (n->type == YAML_SCALAR_NODE)
+    if (n->type == YAML_SCALAR_NODE)/* Is it really used? */
     {
         if (n->data.scalar.length == 0)
         {
-            ERROR(CS_YAML_ERR_PREFIX "found the dependency node to be "
-                  "badly formatted");
+            ERROR(CS_YAML_ERR_PREFIX "the dependency node is badly formatted");
             return TE_EINVAL;
         }
 
-        dep_entry = TE_ALLOC(sizeof(*dep_entry));
-        if (dep_entry == NULL) {
+        obj->depends_count = 1;
+        obj->depends = TE_ALLOC(sizeof(depends_entry));
+
+        if (obj->depends == NULL) {
             ERROR(CS_YAML_ERR_PREFIX "failed to allocate memory");
             return TE_ENOMEM;
         }
 
-        dep_entry->oid = (const xmlChar *)n->data.scalar.value;
-
-        /* Error path resides in parse_config_yaml_cmd_process_target(). */
-        SLIST_INSERT_HEAD(&c->deps, dep_entry, links);
+        obj->depends->oid = strdup((char *)n->data.scalar.value);
     }
     else if (n->type == YAML_SEQUENCE_NODE)
     {
         yaml_node_item_t *item = n->data.sequence.items.start;
+        unsigned int i = 0;
 
+        obj->depends_count = 0;
         do {
-            yaml_node_t *in = yaml_document_get_node(d, *item);
+            obj->depends_count++;
+        } while (++item < n->data.sequence.items.top);
+        obj->depends = TE_ALLOC(obj->depends_count * sizeof(depends_entry));
 
-            dep_entry = TE_ALLOC(sizeof(*dep_entry));
-            if (dep_entry == NULL) {
-                ERROR(CS_YAML_ERR_PREFIX "failed to allocate memory");
-                return TE_ENOMEM;
-            }
+        item = n->data.sequence.items.start;
+        do {
+            yaml_node_t *in = yaml_document_get_node(d, *item);
 
-            rc = parse_config_yaml_dependency_entry(d, in, dep_entry);
+            rc = parse_config_yaml_dependency_entry(d, in, &obj->depends[i]);
             if (rc != 0)
-            {
-                free(dep_entry);
                 return rc;
-            }
 
-            /* Error path resides in parse_config_yaml_cmd_process_target(). */
-            SLIST_INSERT_HEAD(&c->deps, dep_entry, links);
+            i++;
         } while (++item < n->data.sequence.items.top);
     }
     else
     {
-        ERROR(CS_YAML_ERR_PREFIX "found the dependce node to be "
-              "badly formatted");
+        ERROR(CS_YAML_ERR_PREFIX "the dependency node is badly formatted");
         return TE_EINVAL;
     }
 
     return 0;
 }
 
+/**
+ * Process the sequence of target nodes for the specified
+ * command in the given YAML document.
+ *
+ * @param ctx               Current doc context
+ * @param n                 Handle of the target sequence in the given YAML
+ *                          document
+ * @param xn_cmd            Handle of command node in the XML document being
+ *                          created
+ * @param cmd               String representation of command
```

Comment:
Punctuation.

Reply:
It is definitely not my code. Should I remove capital letters in such cases everywhere in conf_yaml.c ?